### PR TITLE
HAAR-1276: Fetch Delius mapping roles from manage users api

### PR DESCRIPTION
**Pull request by ravmoj**
Fetch Delius mapping roles from manage users api

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/ManageUsersApiWebClientConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  fun manageUsersApiWebClient(
+    builder: WebClient.Builder,
+    authorizedClientManager: OAuth2AuthorizedClientManager,
+  ): WebClient = getWebClient(builder, authorizedClientManager)
+
+  @Bean("manageUsersApiUserWebClient")
+  fun manageUsersApiUserWebClient(
+    builder: WebClient.Builder,
+  ): WebClient = getWebClientWithCurrentUserToken(builder)
+
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/ManageUsersApiWebClientConfiguration.kt:27]: Do we need a userWebclient for manage-users?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/AuthSource.kt [hunk 2]
```diff
@@ -3,7 +3,7 @@ package uk.gov.justice.digital.hmpps.oauth2server.security
 import com.fasterxml.jackson.annotation.JsonValue
 
 enum class AuthSource(val description: String) {
-  auth("External"), azuread("Microsoft Azure"), delius("Delius"), nomis("DPS"), none("None"); // ktlint-disable enum-entry-name-case
+  auth("External"), azuread("Microsoft Azure"), delius("Delius"), manage_users_api("Manage users API"), nomis("DPS"), none("None"); // ktlint-disable enum-entry-name-case
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/AuthSource.kt:6]: The authsource is just the users identity provider - not sure if we need manage-users-api here

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt [hunk 3]
```diff
@@ -92,6 +93,7 @@ class UserService(
       nomis -> Optional.ofNullable(nomisUserService.getNomisUserByUsername(username))
       azuread -> azureUserService.getAzureUserByUsername(username)
       delius -> deliusUserService.getDeliusUserByUsername(username)
+      manage_users_api -> Optional.empty()
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt:96]: Don't think is needed if we take it out of authsource

File: helm_deploy/hmpps-auth/values.yaml [hunk 4]
```diff
@@ -91,26 +91,6 @@ generic-service:
     JWT_COOKIE_EXPIRY_TIME: "12h"
     APPLICATION_SMOKETEST_ENABLED: "false"
     SPRING_DATASOURCE_URL: "jdbc:postgresql://${SPRING_DATASOURCE_ENDPOINT}/${SPRING_DATASOURCE_NAME}?sslmode=verify-full"
```
Comment (karenmillermoj) [inline helm_deploy/hmpps-auth/values.yaml:93]: Do we also need to change the other values yml files to add in the manage-users-api spring profile?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/AuthSource.kt (hunk 2 above)
Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/AuthSource.kt]: Updated

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/manage/service/ManageUsersApiService.kt [hunk 5]
```diff
@@ -82,7 +82,7 @@ class ManageUsersApiService(
 
       return deliusRoleList.run { this?.toList() ?: emptyList() }
     } catch (e: ManageUserApiUnreachableServiceException) {
-      ServiceUnavailableThreadLocal.addService(AuthSource.manage_users_api)
+      ServiceUnavailableThreadLocal.addService(delius)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/manage/service/ManageUsersApiService.kt:85]: I think this should still be manageusersapi (sorry, I may have caused some confusion) - so I think this possibly needs to change to take a different param?  Possibly worth speaking to Simon

**Review by karenmillermoj — APPROVED** [review]

**Pull request merged**